athena-vertica/src/main/java/com/amazonaws/connectors/athena/vertica/VerticaMetadataHandler.java

```diff
@@ -0,0 +1,385 @@
+     *
+     * Here we execute the SQL on Vertica
+     *
+     * @param allocator Tool for creating and managing Apache Arrow Blocks.
+     * @param request Provides details of the catalog, database, table, andpartition(s) being queried as well as
+     * any filter predicate.
+     * @return A GetSplitsResponse which primarily contains:
+     * 1. A Set<Split> which represent read operations Amazon Athena must perform by calling your read function.
+     * 2. (Optional) A continuation token which allows you to paginate the generation of splits for large queries.
+     */
+    @Override
+    public GetSplitsResponse doGetSplits(BlockAllocator allocator, GetSplitsRequest request)
+    {
+        //ToDo: implement use of a continuation token to use in case of larger queries
+
+        logger.info("doGetSplits: " + request);
+
+        Set<Split> splits = new HashSet<>();
+        FieldReader fieldReader = request.getPartitions().getFieldReader("preparedStmt");
+
+        String sqlStatement  = fieldReader.readText().toString();
+        String catalogName = request.getCatalogName();
+
+        /**
+         * Build sql based on what query has been issued and concat it with the export to parquet command
+         */
+
+        try{
+            Connection connection = getOrCreateConn(request);
+
+            logger.info("Generated EXPORT statement is: " + sqlStatement);
+
+            //get AWS KEYS from AWS Secrets Manager
+            String awsSecrets = getSecret(System.getenv(AWS_SECRETS));
+
+            //extracting Secrets
+            String replacedSecretString = (awsSecrets.replace("{","")
+                    .replace("}","")
+                    .replace("\"","")
+                    .replaceAll("\\s+",""));
+
+            Map<String, String> result = Arrays.stream(replacedSecretString.split(","))
+                    .map(k ->  k.split(":"))
+                    .collect(Collectors.toMap(
+                            a -> a[0],  //key
+                            a -> a[1]   //value
+                    ));
+
+            String awsSecretKey = result.get("aws_secret_access_key");
+            String awsAccessId = result.get("aws_access_key_id");
+
+            //Generating the sql to set the AWS Session Config on Vertica
+            StringBuilder sb = new StringBuilder();
+            ST temp = new ST("ALTER SESSION SET AWSAuth='<access_key>:<secret_key>'");
+            temp.add("access_key", awsAccessId);
+            temp.add("secret_key", awsSecretKey);
+            String credS = temp.render();
+
+            PreparedStatement credPS = connection.prepareStatement(credS);
+            PreparedStatement exportSQL = connection.prepareStatement(sqlStatement);
+
+            // Set the AWS Region on Vertica
+            String defaultRegion = System.getenv(DEFAULT_REGION).toLowerCase();
+            String awsRegion =  "SELECT SET_CONFIG_PARAMETER('AWSRegion','"+defaultRegion+"') AS REGION_NAME";
```

Comment:
We should move all of this to its own function. Also, is this all being done to provide Vertical with a set of credentials it can use to write to S3? If so, can we discuss this offline. as I have some questions about if customers will be comfortable with doing it this way as these credentials don't seem to expire. Notable, does Vertical support using STS credentials (key, id, token) which can be made to expire periodically and that the Lambda function already can get without the need for  Secretsmanager.

Reply:
Yes. This was to get creds to write to S3. I am working on using STS to get the user authenticated in Vertica. In the meantime, I have moved this to a different function.